test/SILOptimizer/generic_specialization_loops_detection_with_loops.swift

```diff
@@ -0,0 +1,105 @@
+// RUN: %target-swift-frontend -O -emit-sil -Xllvm -sil-print-generic-specialization-loops -Xllvm -sil-print-generic-specialization-info %s 2>&1 | %FileCheck --check-prefix=CHECK %s
+
```

Comment:
Is it guaranteed that this test also tests the inlining case? Maybe you could add an inline(__always), just to be sure.
(And BTW, this is a nice test case!)